playground/src/modules/x/counter/counter.js

```diff
@@ -1,8 +1,35 @@
-import { LightningElement } from 'lwc';
+import { LightningElement, api } from 'lwc';
 
 export default class extends LightningElement {
     counter = 0;
 
+    _foo = 'foo';
+
+    get foo() {
+        return this._foo;
+    }
+
+    set foo(newValue) {
+        this._foo = newValue;
+    }
+
+    _bar = 'bar';
+
+    @api
+    get bar() {
+        return this._bar;
+    }
+
+    set bar(newValue) {
+        this._bar = newValue;
+    }
+
+    _doe = 'doe';
+
+    set doe(newValue) {
+        this._doe = newValue;
+    }
+
```

Comment:
Rather than modifying the playground code (which is designed for quick bug reproductions), could you add an integration test to `packages/@lwc/integration-karma`?

I would recommend creating a brand-new `*.spec.js` file somewhere (don't worry about the directory structure – just put it somewhere that makes sense to you), along with its own `x/` components to reproduce the behavior you're testing. Take a look at other tests in `integration-karma` to get an idea of how this works. There is also a README in that package explaining how to run the tests.

Reply:
I fixed all the other issues. Tomorrow I'm going to write thee tests.

Reply:
@nolanlawson how can I debug an specific karma test in vscode? I'm having 2 errors and I want to iterate over the execution to understand what is happening

![image](https://github.com/salesforce/lwc/assets/55927613/42ac0a35-5e24-4932-9a2c-ae9d71832538)

Reply:
my dist/env.js says tests are running in "development" mode. This means the console.log messages are expected. So, what do you recomend doing with that throwIfConsoleCalled error?
![image](https://github.com/salesforce/lwc/assets/55927613/6a7465cb-f5d3-48ae-8c61-229fdaadca56)

Reply:
@AllanOricil 

> how can I debug an specific karma test in vscode?

To debug a specific test, I usually change `describe` to `fdescribe` or `it` to `fit`. This is a Jasmine convention for [a "focused" test](https://stackoverflow.com/a/23793631). Then I run `yarn start` and then click the "debug" button in the Karma UI, then I open the DevTools to debug.

To test in different modes (e.g. prod vs dev), you can do:

```shell
NODE_ENV_FOR_TEST=production yarn test # or yarn start
NODE_ENV_FOR_TEST=development yarn test # or yarn start
```

> So, what do you recomend doing with that throwIfConsoleCalled error?

That test is specifically designed to catch cases where we are _not_ capturing console messages. If your test is using the typical `expect().toLog*` APIs, then `throwIfConsoleCalled` should not throw, because the console messages will be captured. Here are the available APIs:

https://github.com/salesforce/lwc/blob/1a92c3d3970ce5ee5372df4a88f5b1b78c466c19/packages/%40lwc/integration-karma/helpers/test-utils.js#L330-L335

So for example:

```js
expect(() => {
  // if you call console.warn here in dev mode, the test will pass!
}). toLogWarningDev()
```

Reply:
cool! I was able to run a single test and debug it on the browser :D
<img width="819" alt="image" src="https://github.com/salesforce/lwc/assets/55927613/be41ad06-7fbe-4a50-ae12-0a76fb84ca29">